iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs

```diff
@@ -0,0 +1,136 @@
+﻿using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Devices.Client.Exceptions;
+using Microsoft.Azure.Devices.Shared;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Client.Transport.Amqp
+{
+    internal class AmqpAuthenticationRefresher : IDisposable
+    {
+        static readonly TimeSpan BufferPeriod = TimeSpan.FromSeconds(120);
+        private readonly AmqpCbsLink AmqpCbsLink;
+        private readonly IotHubConnectionString ConnectionString;
+        private readonly string Audience;
```

Comment:
>Audience [](start = 32, length = 8)

Example for audience values. Devicename = z-001

"z-test.azure-devices.net/devices/z-001/messages/events"
"z-test.azure-devices.net/devices/z-001/messages/deviceBound"
"z-test.azure-devices.net/devices/z-001/methods/deviceBound"
"z-test.azure-devices.net/devices/z-001/methods/deviceBound"
"z-test.azure-devices.net/devices/z-001/twin"
"z-test.azure-devices.net/devices/z-001/twin"

Reply:
Go with `endpoint+device id`